addons/event_sale/models/sale_order.py

```diff
@@ -54,6 +55,21 @@ def _compute_attendee_count(self):
         for sale_order in self:
             sale_order.attendee_count = attendee_count_data.get(sale_order.id, 0)
 
+    def _compute_has_soldout_event(self):
+        """Meant to be used in the portal to prevent customers from paying for an event
+        that is sold out and has autoconfirmation. Not only because we'd overbook the
+        event, but also because it will probably crash on the confirmation of the order
+        when the registrations are created."""
+        self.has_soldout_event = False
+        for order in self.filtered(lambda x: x.state in ["draft", "sent"]):
+            order.has_soldout_event = bool(order.order_line.event_id.filtered(
+                lambda x: x.seats_limited and x.seats_max and x.auto_confirm and x.seats_available <= 0
+            ))
```

Comment:
Can't we use _check_seats_availability for this?
Also please use variable names that are somewhat relevant to what they represent, for ease of reading.
e.g. at least o instead of x